src/lsp.rs

```diff
@@ -28,5 +28,111 @@
 //! - Progress notifications - contains a `token` property which could be used to identify the
 //!   client but the specification also says it has nothing to do with the request IDs
 
+use serde_derive::{Deserialize, Serialize};
+
 pub mod jsonrpc;
 pub mod transport;
+
+/// Params for the `initialize` request
+#[derive(Serialize, Deserialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct InitializeParams {
+    pub process_id: Option<u64>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub client_info: Option<ClientInfo>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub locale: Option<String>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub root_path: Option<String>,
+
+    pub root_uri: Option<String>,
+
+    /// User provided initialization options
+    ///
+    /// This is where lspmux-proxy should be inserting it's setup. However we
+    /// should remove them again before forwarding them to the language server.
+    pub initialization_options: Option<InitializationOptions>,
+
+    pub capabilities: serde_json::Value,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub trace: Option<TraceValue>,
+
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub workspace_folders: Vec<WorkspaceFolder>,
+}
+
+#[derive(Serialize, Deserialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct ClientInfo {
+    pub name: String,
+    pub version: Option<String>,
+}
+
+#[derive(Serialize, Deserialize, Clone, Default)]
+pub struct InitializationOptions {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub lsp_mux: Option<LspMuxOptions>,
```

Comment:
This should be `lspMux`, not `lsp_mux`.

```suggestion
#[derive(Serialize, Deserialize, Clone, Default)]
#[serde(rename_all = "camelCase")]
pub struct InitializationOptions {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub lsp_mux: Option<LspMuxOptions>,
```

Reply:
Good catch!